core/app/models/spree/product.rb

```diff
@@ -42,11 +42,24 @@ class Product < Spree::Base
     translates(*TRANSLATABLE_FIELDS)
 
     self::Translation.class_eval do
+      before_update :set_slug
       acts_as_paranoid
       # deleted translation values also need to be accessible for index views listing deleted resources
       default_scope { unscope(where: :deleted_at) }
+      def set_slug
+        if self.name.blank? & self.slug.blank?
+            self.slug = translated_model.translations.find_by(locale: translated_model.stores.first.default_locale).name.to_url
+        else
+          if self.slug.blank?
+            self.slug = self.name.to_url
+          else
+            self.slug = self.slug.to_url
```

Comment:
What does this do? Sets the slug to slug - what's the exact use case?

Reply:
actually, the only use case here is making sure, the slug is in slug format.